ecs/README.md

```diff
@@ -7,121 +7,174 @@ Full instructions can be found in the
 ```
 
-### Create Launch Configuration
+### Deploy hello-world
+
+Now that all our foundational services are deployed and the task definitions are
+available, we can deploy a sample app. The `hello-world` task is composed of a
+`hello` service, a `world` service, and a `world-v2` service. To demonstrate
+inter-service communication, we configure the `hello` service to call the `world`
+service, via `linkerd`.
 
 ```bash
-aws autoscaling create-launch-configuration \
-  --launch-configuration-name l5d-demo-lc \
-  --image-id ami-62e0d802 \
-  --instance-type m4.xlarge \
-  --user-data file://ecs-user-data.txt \
-  --iam-instance-profile ecsInstanceRole \
-  --security-groups $GROUP_ID \
-  --key-name $KEY_PAIR
+aws ecs run-task --cluster l5d-demo --task-definition hello-world --count 2
 ```
 
-### Create Auto Scaling Group
+Note that we have deployed two instances of `hello-world`, which results in two
+`hello` containers, two `world` containers, and two `world-v2` containers.
 
-```bash
-aws autoscaling create-auto-scaling-group \
-  --auto-scaling-group-name l5d-demo-asg \
-  --launch-configuration-name l5d-demo-lc \
-  --min-size 1 --max-size 3 --desired-capacity 2 \
-  --tags ResourceId=l5d-demo-asg,ResourceType=auto-scaling-group,Key=Name,Value=l5d-demo-ecs,PropagateAtLaunch=true \
-  --availability-zones us-west-1a
-```
+## Test everything worked
 
-### Deploy `hello-world`
+First we need to create an SSH tunnel to the cluster. The following commands will
+choose one of the EC2 hosts, and forward traffic on three local ports to three
+remote ports on the EC2 host:
 
-```bash
-aws ecs run-task --cluster l5d-demo --task-definition hello-world --count 2
-```
+- Traffic to `localhost:9990` will go to the Linkerd dashboard on the remote host
+- Traffic to `localhost:8500` will go to the Consul admin dashboard on the remote host
+- Traffic to `localhost:4140` will go to the Linkerd HTTP proxy on the remote host
+- Traffic to `localhost:3000` will go to the Linkerd visualizer on the remote host (will launch this later)
 
-## Test everything worked
+Note that if one of these four ports is already in use on your local machine
+you will either have to stop whatever software if using that port, or you can
+eliminate the port collision by replace the local port number with another unused
+port number in the SSH tunnel command as well as all subsequent commands.
 
 ```bash
 # Select an ECS node
 ECS_NODE=$( \
   aws ec2 describe-instances \
     --filters Name=instance-state-name,Values=running Name=tag:Name,Values=l5d-demo-ecs \
-    --query Reservations[*].Instances[0].PublicDnsName --output text \
+    --query 'Reservations[0].Instances[0].PublicDnsName' \
+    --output text \
 )
 
-# test routing via linkerd
-http_proxy=$ECS_NODE:4140 curl hello
+ssh -f -i "~/.ssh/<your key name here>.pem" \
+    -L 127.0.0.1:4140:$ECS_NODE:4140 \
+    -L 127.0.0.1:9990:$ECS_NODE:9990 \
+    -L 127.0.0.1:3000:$ECS_NODE:3000 \
```

Comment:
since linkerd-viz on port 3000 only runs on one instance, we can't guarantee it will be this particular `ECS_NODE`, mind removing this line?